_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,467 @@
+* *Login settings*: 
+** For the *Valid Redirect URIs*, leave it empty for now. We will come back to edit it later.
+
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to the *Keys* tab for my client from the *Client details* page, 
+* click on the *Import* button located at the bottom of the screen. 
+** Under *Archive format*, choose *PKCS12* from the dropdown. 
+** Under *Key alias*, enter the alias of the RSA key you just created. 
+** Under *Store password* enter the password of the keystore you created.
+** under *Import file*, click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+** Once you select the keystore file from your filesystem, click *Import* and you should see a message at the top of the screen indicating that the certificate has been uploaded successfully and you will see the certificate listed in the text field in the middle of the screen.
+
+. Finally, create a user called *alice* as follows: 
+* Click *Users* in the left hand menu. 
+* Click *Add user*.
+* Fill in the form with the following values:
+ ** *Username*: *alice*.
+ ** *First name*: *Alice*.
+ ** *Last name*: *Smith*. 
+ ** Click *Create*.
+* You can find more details about creating and managing KeyCloak users https://www.keycloak.org/docs/latest/server_admin/#proc-creating-user_server_administration_guide[here].
+
+. This user needs a password to log in. To set the initial password:
+
+* Click *Credentials* at the top of the page.
+* Fill in the *Set password* form with a password.
+* Toggle *Temporary* to *Off* so that the user does not need to update this password at the first login.
+* Hit *Save*.
+
+== Create an OpenShift Secret
+
+Since WildFly will use the keystore we created earlier, we need to add it to OpenShift. We can do this by generating an OpenShift secret using the keystore file as follows: 
+```
+    $ oc create secret generic simple-webapp-secret --from-file=/PATH/TO/Keycloak.keystore.pkcs12
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Add Helm Configuration
+* Obtain the URL for Keycloak.
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak URL:   $KEYCLOAK_URL" &&
+echo ""
+```
+
+* Switch to the charts directory in the `elytron-oidc-client-scope` example.
+```
+    $ cd /PATH/TO/ELYTRON/EXAMPLES/elytron-oidc-request/charts
+```
+Notice there’s a helm.yaml file in this directory with the following content:
+```
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: elytron-oidc-request
+deploy:
+  replicas: 1
+  volumes:
+    - name: saml-keystore-volume
```

Comment:
we should tweak the name since this was the name used for the saml example

Reply:
fixed!